From a5f06e952a71562ab2372123d89e347aaef28d03 Mon Sep 17 00:00:00 2001
From: 1rns <lrns@proton.me>
Date: Wed, 21 Feb 2024 15:29:12 +1300
Subject: [PATCH] fix: ctrlf needs `exit-minibuffer` when it is active

---
 meow-command.el | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/meow-command.el b/meow-command.el
index ec56b82..ce54710 100644
--- a/meow-command.el
+++ b/meow-command.el
@@ -1520,9 +1520,11 @@ Argument ARG if not nil, switching in a new window."
 (defun meow-minibuffer-quit ()
   "Keyboard escape quit in minibuffer."
   (interactive)
-  (if (fboundp 'minibuffer-keyboard-quit)
-      (call-interactively #'minibuffer-keyboard-quit)
-    (call-interactively #'abort-recursive-edit)))
+  (if ctrlf--active-p
+      (call-interactively #'exit-minibuffer)
+    (if (fboundp 'minibuffer-keyboard-quit)
+        (call-interactively #'minibuffer-keyboard-quit)
+      (call-interactively #'abort-recursive-edit))))
 
 (defun meow-escape-or-normal-modal ()
   "Keyboard escape quit or switch to normal state."
-- 
2.43.0

